x/update_validators/controller.go

```diff
@@ -0,0 +1,48 @@
+package update_validators
+
+import (
+	"github.com/confio/weave"
+	"github.com/confio/weave/orm"
+	abci "github.com/tendermint/abci/types"
+)
+
+// Controller is the functionality needed by
+// cash.Handler and cash.Decorator. BaseController
+// should work plenty fine, but you can add other logic
+// if so desired
+type Controller interface {
+	UpdateValidators(store weave.KVStore, diff []abci.Validator)
+}
+
+// BaseController is a simple implementation of controller
+// wallet must return something that supports AsSet
+type BaseController struct {
+	bucket orm.Bucket
+}
+
+// NewController returns a basic controller implementation
+func NewController(bucket orm.Bucket) BaseController {
+	return BaseController{bucket: bucket}
+}
+
+func (c BaseController) UpdateValidators(store weave.KVStore, user weave.Address, diff []abci.Validator) ([]abci.Validator, error) {
```

Comment:
Since diff is never modified when returned (either as is, or with an error), I think `CanUpdateValidators(store, user, diff) error` would be cleaner.